File: c/scrabble-score/scrabble_score.c
#include "scrabble_score.h"

unsigned int score(const char *word) {
    int result = 0;

    if (!word) {
        return result;
    }

    if ((int)strlen(word) == 0) {
        return result;
    }

    while (*word) {
        char letter = (char)tolower(*word++);

        switch (letter) {
        case 'd':
        case 'g':
            result += 2;
            break;
        case 'b':
        case 'c':
        case 'm':
        case 'p':
            result += 3;
            break;
        case 'f':
        case 'h':
        case 'v':
        case 'w':
        case 'y':
            result += 4;
            break;
        case 'k':
            result += 5;
            break;
        case 'j':
        case 'x':
            result += 8;
            break;
        case 'q':
        case 'z':
            result += 10;
            break;
        default:
            // A, E, I, O, U, L, N, R, S, T
            result += 1;
            break;
        }
    }

    return result;
}

File: python/scrabble-score/scrabble_score.py
"""Python Scrabble Score Exercism"""


def score_letter(letter: str) -> int:
    """Scores an individual scrabble letter.

    :param letter: lowercase character
    :result: numeric score
    """

    result: int = 0

    match letter:
        case "a" | "e" | "i" | "o" | "u" | "l" | "n" | "r" | "s" | "t":
            result = 1
        case "d" | "g":
            result = 2
        case "b" | "c" | "m" | "p":
            result = 3
        case "f" | "h" | "v" | "w" | "y":
            result = 4
        case "k":
            result = 5
        case "j" | "x":
            result = 8
        case "q" | "z":
            result = 10
        case _:
            result = 0

    return result


def score(word: str) -> int:
    """Generate the score of a scrabble word.

    :param word: string of letters
    :return: numerical score
    """

    result: int = 0
    chars: list[str] = [*word.lower()]

    result = sum(map(score_letter, chars))

    return result
